File: DatabaseMicroservice/src/main/java/com/licenta/databasemicroservice/presentation/controller/JobController.java
package com.licenta.databasemicroservice.presentation.controller;

import com.licenta.databasemicroservice.business.interfaces.IJobService;
import com.licenta.databasemicroservice.business.model.job.JobRequest;
import com.licenta.databasemicroservice.business.model.job.JobResponse;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.validation.annotation.Validated;
import org.springframework.web.bind.annotation.CrossOrigin;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestMethod;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import javax.validation.Valid;
import javax.validation.constraints.Min;

@Validated
@RestController
@CrossOrigin
@RequestMapping(value="/api/jobs")
public class JobController {

    @Autowired
    private IJobService jobService;

    @RequestMapping(method=RequestMethod.POST)
    public void createJob (@Valid @RequestBody JobRequest request) {

        jobService.createJob(request);
    }

    @RequestMapping(method=RequestMethod.GET)
    public Iterable<JobResponse> getAllJobs(@RequestParam(required = false) Long companyIndustryId) {

        return (companyIndustryId == null) ? jobService.getJobs() : jobService.getCompanyIndustryJobs(companyIndustryId);
    }

    @RequestMapping(value="/{jobId}", method=RequestMethod.GET)
    public JobResponse getJob(@Min(1) @PathVariable Long jobId) {

        return jobService.getJob(jobId);
    }

    @RequestMapping(value="/{jobId}", method=RequestMethod.PUT)
    public void updateJob(@Min(1) @PathVariable Long jobId, @Valid @RequestBody JobRequest request) {

        jobService.updateJob(jobId, request);
    }

    @RequestMapping(value="/{jobId}", method=RequestMethod.DELETE)
    public void deleteJob(@Min(1) @PathVariable Long jobId) {

        jobService.deleteJob(jobId);
    }
}

File: SearchMicroservice/src/main/kotlin/com/licenta/searchmicroservice/presentation/controller/JobController.kt
package com.licenta.searchmicroservice.presentation.controller

import com.licenta.searchmicroservice.business.interfaces.IFilterService
import com.licenta.searchmicroservice.business.model.JobQuery
import com.licenta.searchmicroservice.business.model.JobQueryResponse
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.validation.annotation.Validated
import org.springframework.web.bind.annotation.*


@Validated
@RestController
@CrossOrigin(origins = ["http://localhost:4200"], maxAge = 3600)
@RequestMapping(value = ["/jobs"])
class JobController {

    @Autowired
    private lateinit var jobFilterService: IFilterService

    @GetMapping
    fun getNewsfeed(
        @RequestParam(defaultValue = "true") cached: Boolean,
        @RequestParam(defaultValue = "") title: List<String>,
        @RequestParam(defaultValue = "") cityId: List<Int>,
        @RequestParam(defaultValue = "") countryId: List<Int>,
        @RequestParam(defaultValue = "") companyId: List<Long>,
        @RequestParam(defaultValue = "") industryId: List<Int>,
        @RequestParam(defaultValue = "") companyIndustryId: List<Long>,
        @RequestParam(defaultValue = "") workTypeId: List<Int>,
        @RequestParam(defaultValue = "") jobTypeId: List<Int>,
        @RequestParam(defaultValue = "") experienceLevelId: List<Int>,
        @RequestParam(defaultValue = "") descriptionKeyword: List<String>,
        @RequestParam(defaultValue = "2000-01-01 00:00:00") postedSince: String,
        @RequestParam(defaultValue = "") jobStatusIdList: List<Int>,
        @RequestParam(defaultValue = "0") page: Int,
        @RequestParam(defaultValue = "10") size: Int
    ): JobQueryResponse {

        val jobQuery = JobQuery(
            title,
            cityId,
            countryId,
            companyId,
            industryId,
            companyIndustryId,
            workTypeId,
            jobTypeId,
            experienceLevelId,
            descriptionKeyword,
            postedSince,
            jobStatusIdList
        )

        return jobFilterService.getFilteredJobs(jobQuery, cached, page, size)
    }
}